pkg/aks/client.go

```diff
@@ -133,5 +142,44 @@ func GetSecrets(secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfi
 	cred.AuthBaseURL = spec.AuthBaseURL
 	cred.BaseURL = spec.BaseURL
 
+	if cred.TenantID == "" {
+		cred.TenantID, err = GetCachedTenantID(secretClient, cred.SubscriptionID, secret)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return &cred, nil
 }
+
+type secretClient interface {
+	Update(*v1.Secret) (*v1.Secret, error)
+}
```

Comment:
Is this type necessary?

Reply:
I'm adding it to try to make this usable both here and by rancher. Here we use `SecretClient` from wrangler, there we have `SecretInterface` from norman, and trying to cast between the two or add new dependencies to one or the other seemed like a headache.

Reply:
Ah, that makes sense!